### ISSUE TITLE
Streamline build process

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#40]
When we separated `make download` from `make all` we removed some lines that set all of our repos to the `develop` branch.

It would be nice to have all the repos set to `develop` at the end of `make download` again so we don't have to revert them manually.

**Comment by rennergade** [comment]
@jesings fixed up our make process so download works coherently! Thanks!

The last step here is to make the environment variables persist in Docker. This includes the variables for `lindsh` as well as the main Lind environment variables (the ones listed in the Making Lind from Scratch wiki).

Once that is done this might actually be sane for someone to come in with no knowledge and set up!

**Comment by rennergade** [comment]
Ok, so in order to do this we need to realistically do a big merge of develop branches to master. There's a number of inconsistencies here that are either are relevant or annoying that should be fixed.

**Lind Project** 

Just needs a straight develop to master merge

**Native Client**
same develop-> master, but seems to have master ahead by one commit?

**NaCl Repy**
Huge discrepancy between the develop and master branches, some of this has to do with commits to the file structure.

**Lind Glibc**
This seems like a straight develop to master... but its forked from a cmatthew repo that I'm not sure we need to keep the reference to?

**NaCl-GCC**
This is also straight develop to master, but master is named i686_caging now? Id like to revert that back to master

**NaCl Binutils**
Develop is way behind master because of inconsistencies in where we took the updates from? I think here we need to make a new "master" branch and put where we are with develop there.

**third party**
Minor inconsistency between develop and master?

**Lind Misc**
Develop and master are at the same point, but has the same cmathew fork issue as glibc

**Reference Monitor**
This is unused and I think is worth removing.

**Comment by rennergade** [comment]
**Third Party**

This is actually strange, our builds aren't even pointed to the repo we have hosted for `third_party`. It's pointed directly at googles [native client repo for third party](https://chromium.googlesource.com/native_client/src/third_party.git) at the latest commit. 

Since this all has been working, I'm pretty sure the small handful of commits to the repo we have are irrelevant. I possibly should archive their third_party repo as our own and get rid of what we currently have.

**Comment by rennergade** [comment]
**nacl_repy**

The branches here are pretty fubar (see image below) and I don't think an actual *merge* is going to be helpful. I think a direct replacement of the master branch with what we currently have in develop makes the most sense.

![image (1)](https://user-images.githubusercontent.com/7719272/74573594-0bf77980-4f4f-11ea-9ba8-f6aa936f9f63.png)

**Comment by rennergade** [comment]
The merge is finished. We can now just pull from docker and immediately have a working image with no shenanigans. Hooray!
